nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-reporting-task/src/test/java/org/apache/nifi/reporting/prometheus/TestPrometheusReportingTask.java

```diff
@@ -0,0 +1,92 @@
+import org.apache.nifi.util.MockVariableRegistry;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestPrometheusReportingTask {
+    private static final String TEST_INIT_CONTEXT_ID = "test-init-context-id";
+    private static final String TEST_INIT_CONTEXT_NAME = "test-init-context-name";
+    private static final String TEST_TASK_ID = "test-task-id";
+    private MockReportingInitializationContext reportingInitContextStub;
+    private MockReportingContext reportingContextStub;
+    private MockConfigurationContext configurationContextStub; // new
+    private PrometheusReportingTask testedReportingTask;
+    private ProcessGroupStatus rootGroupStatus;
+
+    @Before
+    public void setup() {
+        testedReportingTask = new PrometheusReportingTask();
+        rootGroupStatus = new ProcessGroupStatus();
+        reportingInitContextStub = new MockReportingInitializationContext(TEST_INIT_CONTEXT_ID, TEST_INIT_CONTEXT_NAME, new MockComponentLog(TEST_TASK_ID, testedReportingTask));
+
+        reportingContextStub = new MockReportingContext(Collections.emptyMap(), new MockStateManager(testedReportingTask), new MockVariableRegistry());
+
+        reportingContextStub.setProperty(PrometheusReportingTask.INSTANCE_ID.getName(), "localhost");
+
+        configurationContextStub = new MockConfigurationContext(reportingContextStub.getProperties(), reportingContextStub.getControllerServiceLookup());
+
+        rootGroupStatus.setId("1234");
+        rootGroupStatus.setFlowFilesReceived(5);
+        rootGroupStatus.setBytesReceived(10000);
+        rootGroupStatus.setFlowFilesSent(10);
+        rootGroupStatus.setBytesSent(20000);
+        rootGroupStatus.setQueuedCount(100);
+        rootGroupStatus.setQueuedContentSize(1024L);
+        rootGroupStatus.setBytesRead(60000L);
+        rootGroupStatus.setBytesWritten(80000L);
+        rootGroupStatus.setActiveThreadCount(5);
+        rootGroupStatus.setName("root");
+        rootGroupStatus.setFlowFilesTransferred(5);
+        rootGroupStatus.setBytesTransferred(10000);
+        rootGroupStatus.setOutputContentSize(1000L);
+        rootGroupStatus.setInputContentSize(1000L);
+        rootGroupStatus.setOutputCount(100);
+        rootGroupStatus.setInputCount(1000);
+
+    }
+
+    @Test
+    public void testOnTrigger() throws IOException, InterruptedException, InitializationException {
+        testedReportingTask.initialize(reportingInitContextStub);
+        testedReportingTask.onScheduled(configurationContextStub);
+        reportingContextStub.getEventAccess().setProcessGroupStatus(rootGroupStatus);
+        testedReportingTask.onTrigger(reportingContextStub);
+
+        URL url = new URL("http://localhost:9092/metrics");
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("GET");
+        int status = con.getResponseCode();
+        Assert.assertEquals(HttpURLConnection.HTTP_OK, status);

```

Comment:
It would be nice to have some assertions on the results as well.

Reply:
added assertion for result.